sdk/src/main/java/io/radar/sdk/RadarVerificationManager.kt

```diff
@@ -18,10 +29,322 @@ internal class RadarVerificationManager(
+                                                    ) {
+                                                        if (status != Radar.RadarStatus.SUCCESS || beacons == null) {
+                                                            callTrackApi(null)
+
+                                                            return
+                                                        }
+
+                                                        callTrackApi(beacons)
+                                                    }
+                                                })
+                                        } else {
+                                            callTrackApi(null)
+                                        }
+                                    }
+                                },
+                                false
+                            )
+                        } else {
+                            callTrackApi(null)
+                        }
+                    }
+                }
+            })
+    }
+
+    fun trackVerifiedToken(beacons: Boolean = false, callback: Radar.RadarTrackTokenCallback? = null) {
+        val verificationManager = this
+
+        val googlePlayProjectNumber = RadarSettings.getGooglePlayProjectNumber(this.context)
+
+        Radar.locationManager.getLocation(
+            RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy.HIGH,
+            Radar.RadarLocationSource.FOREGROUND_LOCATION,
+            object :
+                Radar.RadarLocationCallback {
+                override fun onComplete(
+                    status: Radar.RadarStatus,
+                    location: Location?,
+                    stopped: Boolean
+                ) {
+                    if (status != Radar.RadarStatus.SUCCESS || location == null) {
+                        Radar.handler.post {
+                            callback?.onComplete(status)
+                        }
+
+                        return
+                    }
+
+                    val requestHash = verificationManager.getRequestHash(location)
+
+                    verificationManager.getIntegrityToken(
+                        googlePlayProjectNumber,
+                        requestHash
+                    ) { integrityToken, integrityException ->
+                        val callTrackApi = { beacons: Array<RadarBeacon>? ->
+                            Radar.apiClient.track(
+                                location,
+                                RadarState.getStopped(verificationManager.context),
+                                RadarActivityLifecycleCallbacks.foreground,
+                                Radar.RadarLocationSource.FOREGROUND_LOCATION,
+                                false,
+                                beacons,
+                                true,
+                                integrityToken,
+                                integrityException,
+                                true,
```

Comment:
Not necessary to change, but this would be easier to follow if we were using named params here.